readthedocs/projects/tasks.py

```diff
@@ -1387,11 +1387,12 @@ def warn(self, msg):
 
 def clean_build(version_pk):
     """Clean the files used in the build of the given version."""
-    version = Version.objects.get_object_or_log(pk=version_pk)
-    if (
-        not version or
-        not version.project.has_feature(Feature.CLEAN_AFTER_BUILD)
-    ):
+    try:
+        version = SyncRepositoryMixin.get_version(version_pk)
```

Comment:
A comment about why this is not using `get_object_or_log` as the rest of the functions here would be good to avoid the same confusion in the future.